common/dm.c

```diff
@@ -83,3 +84,24 @@ dm_init(void)
 	for (struct device *dev = device_list; dev < device_list_end; ++dev)
 		device_probe(dev);
 }
+
+int
+dm_setup_pins(struct device *dev, uint8_t num_pins)
+{
+	struct gpio_handle *pins = dev->gpio_pins;
+	struct device *gpio_dev;
+	uint8_t pin_id, mode;
+	int     err;
+
+	for (size_t i = 0; i < num_pins; ++i) {
+		gpio_dev = pins[i].dev;
+		pin_id   = pins[i].pin;
+		mode     = pins[i].mode;
+
+		/* Set pin mode and return if error occurs. */
+		if ((err = gpio_set_mode(gpio_dev, pin_id, mode)))
```

Comment:
Call `device_probe` on `gpio_dev` before calling `gpio_set_mode`, to make sure the driver for the GPIO controller is loaded.

Reply:
Done.